src/System.Net.Http/src/Internal/Mail/MailBnfHelper.cs

```diff
@@ -330,9 +333,21 @@ internal static string GetDateTimeString(DateTime value, StringBuilder builder)
                 localBuilder.Append(' ');
             }
 
-            string[] offsetFields = offset.Split(':');
-            localBuilder.Append(offsetFields[0]);
-            localBuilder.Append(offsetFields[1]);
+            int colonIndex = offset.IndexOf(':');
+            if (colonIndex == -1)
+            {
+                throw new InvalidOperationException("TimeSpan.ToString() should contain at least one colon.");
```

Comment:
> I don't think this code is actually being used anywhere else in the repo

Then what's the value in changing it?  We should only be churning code to improve performance in places where it matters, and it doesn't matter in code that's not being used.